rcl/include/rcl/time.h

```diff
@@ -24,9 +24,9 @@ extern "C"
 #include "rcl/types.h"
 #include "rcl/visibility_control.h"
 
-#define RCL_S_TO_NS(seconds) (seconds * (1000 * 1000 * 1000))
-#define RCL_MS_TO_NS(milliseconds) (milliseconds * (1000 * 1000))
-#define RCL_US_TO_NS(microseconds) (microseconds * 1000)
+#define RCL_S_TO_NS(seconds) ((int64_t)seconds * (1000 * 1000 * 1000))
+#define RCL_MS_TO_NS(milliseconds) ((int64_t)milliseconds * (1000 * 1000))
+#define RCL_US_TO_NS(microseconds) ((int64_t)microseconds * 1000)
```

Comment:
This cannot be cast to `uint64_t` though, because if the given value is negative it shouldn't be cast to an unsigned type.

I would revert this and ensure the temporary storage is large enough at the point of use, as in the lines referred to @dhood: https://github.com/ros2/rcl/pull/86/files#diff-eebe14c889bc8ee26a92052b7ae5ae08L69